graphrag/index/workflows/v1/subflows/create_final_text_units_pre_embedding.py

```diff
@@ -0,0 +1,178 @@
+                "to": "text",
+            },
+            {
+                "column": "n_tokens",
+                "operation": "any",
+                "to": "n_tokens",
+            },
+            {
+                "column": "document_ids",
+                "operation": "any",
+                "to": "document_ids",
+            },
+            {
+                "column": "entity_ids",
+                "operation": "any",
+                "to": "entity_ids",
+            },
+            {
+                "column": "relationship_ids",
+                "operation": "any",
+                "to": "relationship_ids",
+            },
+            *(
+                []
+                if not covariates_enabled
+                else [
+                    {
+                        "column": "covariate_ids",
+                        "operation": "any",
+                        "to": "covariate_ids",
+                    }
+                ]
+            ),
+        ],
+        ["id"],
+    )
+
+    return create_verb_result(aggregated)
+
+
+def _entities(table):
+    selected = cast(Table, table[["id", "text_unit_ids"]])
+    unrolled = selected.explode("text_unit_ids").reset_index(drop=True)
+    return aggregate_df(
+        unrolled,
+        [
+            {
+                "column": "id",
+                "operation": "array_agg_distinct",
+                "to": "entity_ids",
+            },
+            {
+                "column": "text_unit_ids",
+                "operation": "any",
+                "to": "id",
+            },
+        ],
+        ["text_unit_ids"],
+    )
+
+
+def _relationships(table):
+    selected = cast(Table, table[["id", "text_unit_ids"]])
+    unrolled = selected.explode("text_unit_ids").reset_index(drop=True)
+    aggregated = aggregate_df(
+        unrolled,
+        [
+            {
+                "column": "id",
+                "operation": "array_agg_distinct",
+                "to": "relationship_ids",
+            },
+            {
+                "column": "text_unit_ids",
+                "operation": "any",
+                "to": "id",
+            },
+        ],
+        ["text_unit_ids"],
+    )
+    return aggregated[["id", "relationship_ids"]]
+
+
+def _covariates(table):
+    selected = cast(Table, table[["id", "text_unit_id"]])
+    return aggregate_df(
+        selected,
+        [
+            {
+                "column": "id",
+                "operation": "array_agg_distinct",
+                "to": "covariate_ids",
+            },
+            {
+                "column": "text_unit_id",
+                "operation": "any",
+                "to": "id",
+            },
+        ],
+        ["text_unit_id"],
+    )
+
+
+def _join(left, right):
+    return __clean_result(
+        left.merge(
+            right,
+            left_on="id",
+            right_on="id",
+            how="left",
+            suffixes=cast(Suffixes, ["_1", "_2"]),
+            indicator=True,
```

Comment:
I think we should only use this if we intend to do some distinction between rows origins, but since at the ends we are doing another concat we could just ignore this.